types/midi/index.d.ts

```diff
@@ -0,0 +1,47 @@
+// Type definitions for midi 2.0
+// Project: https://github.com/justinlatimer/node-midi
+// Definitions by: Ryan Price <https://github.com/ryprice>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare namespace MIDI {
+    type MIDIMessage = [number, number, number];
+
+    type MIDIMessageHandler = (
+        deltaTime: number,
+        // [status, data1, data2]
+        message: MIDIMessage,
+    ) => void;
+
+    interface MIDIInput {
+        getPortCount(): number;
+        getPortName(port: number): string;
+        on(type: 'message', handler: MIDIMessageHandler): void;
+        openPort(port: number): void;
+        ignoreTypes(filterSysex: boolean, filterTiming: boolean, filterSensing: boolean): void;
+    }
+
+    interface MIDIInputConstructor {
+        new (): MIDIInput;
+    }
+
+    interface MIDIOutput {
+        getPortCount(): number;
+        getPortName(port: number): string;
+        openPort(port: number): void;
+        sendMessage(message: MIDIMessage): void;
+        closePort(): void;
+    }
+
+    interface MIDIOutputConstructor {
+        new(): MIDIOutput;
+    }
+
+    interface MIDIStatic {
+        Input: MIDIInputConstructor;
+        Output: MIDIOutputConstructor;
+    }
+
+    const MIDIStatic: MIDIStatic;
+}
+
+export = MIDI.MIDIStatic;
```

Comment:
I suggest

```suggestion
// Type definitions for midi 2.0
// Project: https://github.com/justinlatimer/node-midi
// Definitions by: Ryan Price <https://github.com/ryprice>
// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped

/// <reference types="node" />

import { EventEmitter, Readable, Writable } from 'stream';

declare namespace MIDI {
    type Message = [number, number, number];

    type MessageHandler = (
        deltaTime: number,
        // [status, data1, data2]
        message: Message,
    ) => void;
}

declare class MIDI extends EventEmitter {
    constructor();

    getPortCount(): number;
    getPortName(port: number): string;
    openPort(port: number): void;
    closePort(): void;
    openVirtualPort(name: string): void;
}

declare class MIDIInput extends MIDI {
    on(eventName: 'message', listener: MIDI.MessageHandler): this;
    ignoreTypes(filterSysex: boolean, filterTiming: boolean, filterSensing: boolean): void;
}

declare class MIDIOutput extends MIDI {
    sendMessage(message: MIDI.Message): void;
}

declare const midi: {
    Input: typeof MIDIInput;
    Output: typeof MIDIOutput;

    /**
     * Backwards compatibility.
     * @see {@link https://github.com/justinlatimer/node-midi/blob/master/midi.js#L9}
     */
    input: typeof MIDIInput;
    output: typeof MIDIOutput;

    /**
     * @see {@link https://github.com/justinlatimer/node-midi/blob/master/midi.js#L14}
     */
    createReadStream(input?: MIDIInput): Readable;
    /**
     * @see {@link https://github.com/justinlatimer/node-midi/blob/master/midi.js#L45}
     */
    createWriteStream(output?: MIDIOutput): Writable;
};

export = midi;
```

Reply:
Thanks for the suggestions. Like I said in the PR description, I didn't write a complete type definition because I'd prefer not to add definitions for code I can't test. Is this generated from a script? I guess it would be ok not to test if it's generated.

Reply:
FYI MIDIOutput extends MIDI extends EventEmitter, but it seems like [only MIDIInput extends EventEmitter](https://github.com/justinlatimer/node-midi/blob/master/midi.js#L5). I'm not sure how to correctly represent this and also use your shared MIDI parent class idea. (I dug and you are right that they [do share a common ancestor](https://github.com/thestk/rtmidi/blob/master/RtMidi.h#L6))

Reply:
> 

No, it is not generated from a script. It is my own definition.

Reply:
> FYI it seems like [only MIDIInput extends EventEmitter](https://github.com/justinlatimer/node-midi/blob/master/midi.js#L5). I'm not sure how to correctly represent this and also use your shared MIDI parent class idea. (I dug and you are right that they [do share a common ancestor](https://github.com/thestk/rtmidi/blob/master/RtMidi.h#L6))

I think you are right, only MIDIInput extends EventEmiter.